bdns/problem.go

```diff
@@ -43,8 +94,54 @@ func (d Error) Error() string {
 	} else {
 		detail = detailServerFailure
 	}
-	return fmt.Sprintf("DNS problem: %s looking up %s for %s%s", detail,
-		dns.TypeToString[d.recordType], d.hostname, additional)
+
+	if d.extended == nil {
+		return fmt.Sprintf("DNS problem: %s looking up %s for %s%s", detail,
+			dns.TypeToString[d.recordType], d.hostname, additional)
+	}
+
+	// A copy of bdns's mapping of error codes to strings. We tweak it slightly so all DNSSEC-related
+	// errors say "DNSSEC" at the beginning.
+	// https://pkg.go.dev/github.com/miekg/dns#ExtendedErrorCodeToString
+	// Also note that not all of these codes can currently be emitted by Unbound. See Unbound's
+	// announcement post for EDE: https://blog.nlnetlabs.nl/extended-dns-error-support-for-unbound/
+	extendedErrorCodeToString := map[uint16]string{
```

Comment:
It feels inefficient to re-declare this map every time this function is called. I might hope that the go optimizer would be smart enough to pull this out into a const, but I don't know how to check if it actually does that.